content/en/blog/_posts/2019-06-13-volume-cloning-alpha.md

```diff
@@ -0,0 +1,107 @@
+
+Instead, Kubernetes users are now empowered to incorporate clone operations in a cluster agnostic way into their tooling and policy with the comfort of knowing that it will work against arbitrary Kubernetes clusters regardless of the underlying storage.
+
+## Which volume plugins support Kubernetes Cloning?
+
+Kubernetes supports three types of volume plugins: in-tree, Flex, and [Container Storage Interface](https://github.com/container-storage-interface/spec/blob/master/spec.md) (CSI). See [Kubernetes Volume Plugin FAQ](https://github.com/kubernetes/community/blob/master/sig-storage/volume-plugin-faq.md) for details.
+
+Cloning is only supported for CSI drivers (not for in-tree or Flex). To use the Kubernetes cloning feature, ensure that a CSI Driver that implements cloning is deployed on your cluster.
+For a list of CSI drivers that currently support cloning see the [CSI Drivers doc](https://kubernetes-csi.github.io/docs/drivers.html).
+
+## Kubernetes API and Cloning
+
+The cloning feature in Kubernetes is enabled by a new `PersistentVolumeClaim.DataSource` field that references an existing PersistentVolumeClaim object in the same namespace.  There are no new objects introduced to enable cloning, instead the existing DataSource field in the PersistentVolumeClaim object is expanded to be able to accept the name of an existing PersistentVolumeClaim in the users namespace.  It is important to note that from a users perspective a clone is just another PersistentVolume and PersistentVolumeClaim, the only difference being that that PersistentVolume is being populated with the contents of another PersistentVolume at creation time.  After creation it behaves exactly like any other Kubernetes PersistentVolume and adheres to the same behaviors and rules.
+
+## Kubernetes Cloning Requirements
+
+Before using Kubernetes Volume Cloning, you must:
+
+* Ensure a CSI driver implementing Cloning is deployed and running on your Kubernetes cluster.
+* Enable the Kubernetes Volume Cloning feature via new Kubernetes feature gate (disabled by default for alpha):
+  * Set the following flag on the API server binary: `--feature-gates=VolumePVCDataSource=true`
+
+## Creating a clone with Kubernetes
+
+To provision a new volume pre-populated with data from an existing Kubernetes Volume, use the dataSource field in the `PersistentVolumeClaim`.  There are three parameters:
+
+* name - name of the `PersistentVolumeClaim` object to use as source
+* kind - must be `PersistentVolumeClaim`
+* apiGroup - must be `""`
+
+The namespace of the source `PersistentVolumeClaim` (source) object is assumed to be the same as the namespace of the new `PersistentVolumeClaim` (destination) object.
```

Comment:
Perhaps better to use the phrase "The namespace of the destination will default to the namespace of the source". "is assumed to be" doesn't really sound right, IMHO.

Also, perhaps worth adding a note about whether cross-namespace cloning will ever be a thing -- i.e. transferring a cloned volume between namespaces?

Reply:
Well hello @jaypipes thanks for the review!  I'm not sure about putting in info on cross-namespace in the release blog (yet).  I'm working on a new KEP to propose a general API if you're interested.  Original proposal (scrapped, but has the info on what we're moving towards):  https://github.com/kubernetes/enhancements/pull/643

Reply:
You could maybe mention this in the "Future" section as something that's being investigated.

Reply:
Sounds good to me, ya'll know I'm a proud supporter of namespace transfer anyway.